api/graphql/types/recurring_reservation/serializers.py

```diff
@@ -38,52 +59,316 @@ class Meta:
+                    end_datetime=end,
+                    is_reservable=True,
+                    buffer_time_after=instance.reservation_unit.buffer_time_after,
+                    buffer_time_before=instance.reservation_unit.buffer_time_before,
+                )
+
+                # Would the reservation overlap with any existing reservations?
+                # Also checks that buffers for the reservation will also not overlap.
+                if any(timespan.overlaps_with(reserved) for reserved in reserved_timespans):
+                    overlapping.append(
+                        ReservationPeriodJSON(
+                            begin=begin.isoformat(timespec="seconds"),
+                            end=end.isoformat(timespec="seconds"),
+                        )
+                    )
+                    continue
+
+                # Would the reservation be fully inside any reservable timespans for the resource?
+                # Ignores buffers for the reservation, since those can be outside reservable times.
+                if check_opening_hours and not any(
+                    timespan.fully_inside_of(reservable) for reservable in reservable_timespans
+                ):
+                    not_reservable.append(
+                        ReservationPeriodJSON(
+                            begin=begin.isoformat(timespec="seconds"),
+                            end=end.isoformat(timespec="seconds"),
+                        )
+                    )
+                    continue
+
+                non_overlapping.append(ReservationPeriod(begin=begin, end=end))
+
+        if overlapping:
+            msg = "Not all reservations can be made due to overlapping reservations."
+            extensions = {"code": error_codes.RESERVATION_SERIES_OVERLAPS, "overlapping": overlapping}
+            raise GraphQLError(msg, extensions=extensions)
+
+        if not_reservable:
+            msg = "Not all reservations can be made due to falling outside reservable times."
+            extensions = {"code": error_codes.RESERVATION_SERIES_NOT_OPEN, "not_reservable": not_reservable}
+            raise GraphQLError(msg, extensions=extensions)
+
+        # Pick out the through model for the many-to-many relationship and use if for bulk creation
+        ThroughModel: type[models.Model] = Reservation.reservation_unit.through  # noqa: N806
+
+        reservations: list[Reservation] = []
+        through_models: list[models.Model] = []
+
+        for period in non_overlapping:
+            reservation = Reservation(
+                name=instance.name,
+                description=instance.description,
+                begin=period["begin"],
+                end=period["end"],
+                user=reservation_user,
+                type=reservation_type,
+                recurring_reservation=instance,
+                age_group=instance.age_group,
+                buffer_time_before=instance.reservation_unit.buffer_time_before,
```

Comment:
Buffer times should be according to request payload (UI defaults now to 0 but allows handler to select to use values which in practice are those defined on the reservation unit).

Reply:
Only request payload, so shouldn't use reservation unit values if buffers are omitted from the request?

Reply:
Hmm... _StaffCreate_ (and _StaffAdjust_) says "Null/undefined value means buffer from reservation unit is used" but that only seems to apply for the validations, not what is in the end saved on the reservation... and when creating the series of reservations UI does not send zero values now but omits them from the request (only when adjusting the time of an existing reservation as staff member zero value is explicitly passed for buffer if it is not wanted on the reservation, but that is necessary so the old values that applied to the old version of the reservation are not kept).

Of course that could be changed and the UI needs to be updated anyway... but since the default scenario is that for staff created reservations buffers are not wanted, can be so that omitting them won't default to the res.unit settings and instead request payload needs to define the buffer values if they are wanted.